espnet2/speechlm/dataloader/multimodal_loader/audio_loader.py

```diff
@@ -17,6 +31,100 @@
     )
 
 
+class ArkiveAudioReader:
+    """Dict-like lazy audio reader using arkive parquets
+
+    This reader supports both single-channel and multi-channel audio data:
+    - Single-channel audio (MonoCut): Returns shape [1, num_samples]
+    - Multi-channel audio (MultiCut): Returns shape [num_channels, num_samples]
+
+    The output shape is consistent regardless of the input type, always returning
+    a 2D array with shape [num_channels, num_samples].
+
+    Args:
+        query: SQL query to access the parquet(s). Must return the required metadata
+            columns: [utt_id, path, start_byte_offset, file_size_bytes, start_time, end_time]
+        valid_ids: List of valid IDs to keep (optional, keeps all if None)
+        worker_id: partition ids by worker (optional, keeps all if None)
+        world_size: used for worker partitioning
+    """
+
+    def __init__(
+        self,
+        query: str,
+        valid_ids: list = None,
+        worker_id: int = None,
+        world_size: int = None,
+    ):
+
+        result = duckdb.query(query)
+
+        # filter query result before loading to df
+        # avoids loading the whole query result into memory
+        if worker_id is not None:
+            assert (
+                world_size is not None
+            ), f"filtering by worker_id requires world_size, got {world_size}"
+            result = duckdb.query(
+                f"""
+                SELECT * FROM result
+                QUALIFY (row_number() OVER (ORDER BY utt_id) - 1) % {world_size} = {worker_id}
+            """
+            )
+
+        df = result.df()
+
+        data = dict(
+            zip(
+                df["utt_id"],
+                zip(
+                    df["path"],
+                    df["start_byte_offset"],
+                    df["file_size_bytes"],
+                    df["start_time"],
+                    df["end_time"],
+                ),
+            )
+        )
+
+        if valid_ids:
+            data = {k: data[k] for k in set(valid_ids) if k in data}
+
+        self.data = data
+
+    def __getitem__(self, key: str) -> Tuple[np.ndarray, int]:
+        path, start_byte, file_size, start_time, end_time = self.data[key]
```

Comment:
`start_time` and `end_time` might be "<NA>" in pandas notation if not provide segments, so here we should convert them to None so that the subsequent code can correctly recognize and handle them:
```suggestion
path, start_byte, file_size, start_time, end_time = self.data[key]
        
        # Convert pandas NA to Python None
        if pd.isna(start_time):
            start_time = None
        if pd.isna(end_time):
            end_time = None
            
```